pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,16 +465,56 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 	if err != nil {
 		return nil, err
 	}
-	for _, item := range items {
-		accessor, err := meta.Accessor(item)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := e.Delete(ctx, accessor.GetName(), options); err != nil && !kubeerr.IsNotFound(err) {
-			return nil, err
+	// Spawn a number of goroutines, so that we can issue requests to etcd
+	// in parallel to speed up deletion.
+	workersNumber := e.DeleteCollectionWorkers
```

Comment:
add a TODO to make this proportional to the number of items to delete, up to the max... feels silly to spawn 16 workers to delete 10 items